### PR TITLE
Pileks/feat/multidocs

### PR DESCRIPTION
**Pull request by pileks**
Includes changes to `Quick Start` and `CLI`, adding support for all of our client languages.

Prerequisites:
- https://github.com/polywrap/rust-client/pull/244
- https://github.com/polywrap/rust-client/pull/245

Remaining tasks:
- [ ] Once prerequisites are merged, we can link to `polywrap` on `docs.rs` for the Reference documentation in the "Clients" section

### CODE REVIEW

File: src/docs/quick-start/rust.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+```
+ba5a5d5fb7674f5975f0ecd0cd9a2f4bcadc9c04f5ac2ab3a887d8f10355fc38
+```
+
+This is the return value of our invocation.
+
+#### What's going on here?
+
+Using the Polywrap Client, we are invoking the `sha3_256` method of a Wrap found under the [WRAP URI](/concepts/uris) `wrapscan.io/polywrap/sha3@1.0` called the SHA3 Wrap.
```
Comment (dOrgJelli) [inline src/docs/quick-start/rust.md:65]: NIT: we can reduce the amount of duplicate text in these docs by using snippets for the contents (and not just the code)

**Review by dOrgJelli — APPROVED** [review]

**Comment by dOrgJelli** [comment]
3 more NITs:
- Somewhere in the quick start pages, I think we should have a call-out saying something like "typings for wraps can be generated by the CLI so their developer experience is 1:1 with traditional SDKs" and link to the CLI section where we codegen the types.
- The application typings within the CLI page aren't all up-to-date. If you look at the different languages, they're all ab it different. I think we should unify these examples.
- I think the CLI page looks great, but it feels weird that it only covers integrating wraps into apps. I think we might want to also cover building wraps as well. Maybe these two use-cases for the CLI should be covered in 2 different pages.

All of this could totally be done in a separate PR. I overall love the look and feel of everything :fire: let's keep iterating and improving on this! It feels much farther along than before!

File: src/docs/quick-start/rust.md (hunk 1 above)
Comment (pileks) [reply src/docs/quick-start/rust.md:65]: I'm actually looking into markdown partials for this kind of thing!

No need to use snippets when Docusaurus' MDX capabilities allow for partial views.

**Comment by pileks** [comment]
> * Somewhere in the quick start pages, I think we should have a call-out saying something like "typings for wraps can be generated by the CLI so their developer experience is 1:1 with traditional SDKs" and link to the CLI section where we codegen the types.

Agreed 100%

> * The application typings within the CLI page aren't all up-to-date. If you look at the different languages, they're all ab it different. I think we should unify these examples.

Do you mean the wraps we're importing? It does seem that, from what I've gathered talking to you guys, we should probably actually unify all template apps to import the same wrap/do the same thing (maybe even something as dumb as SHA3 invocation).

I think that some of the app templates currently are a bit "gratuitous" in the sense that they are basically full apps instead of just giving people a good starting point with a simple example. We can even keep the more complex templates, as we do have support for pulling templates from non-default sources.

I believe such a move (unified starter examples) would allow us to unify our documentation more, and what are templates but a live documentation example.
We can also leave all our current, more complex templates, and write in the docs how people can pull those.

Let me know what your thoughts on this are and maybe we can discuss this further with all other core devs on ACDC.

> * I think the CLI page looks great, but it feels weird that it only covers integrating wraps into apps. I think we might want to also cover building wraps as well. Maybe these two use-cases for the CLI should be covered in 2 different pages.

Huh, so actually adding a "more advanced" tutorial here. I'm generally OK with that, at least for a very simple example. The thing is that we'd also have to add the concept of sub-invocation here (which shouldn't be too hard to explain).
See in our Notion: https://www.notion.so/polywrap/Multi-language-Docs-e23e3a223b4c4853806e2c0dd48f1981?pvs=4#0f82054e875648589e1dfee300a29f86
There's the "Wrap language specific pages" which essentially cover building Wraps. We could think about giving users an intro to the `create wasm` command and linking them to the tutorials, once I manage to write them out.

**Pull request merged**